### PR TITLE
server: event for HA vm start

### LINKED ISSUE #8021: Missing event after ha restart of VM
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.18.1.0
~~~

##### SUMMARY

When using a ha enabled offering and a VM was restarted because of this, no entry is showing in events.

##### STEPS TO REPRODUCE

~~~
1) Create a VM with ha enabled offering
2) Destroy the VM on the hypervisor (in KVM via virsh destroy <id>)
3) Wait until CS is restarting the VM
4) Take a look at Events tab of VM and/or Events Tab of main menu
~~~

##### EXPECTED RESULTS

~~~
A new entry should be added to the events history because of ha restart
~~~

##### ACTUAL RESULTS

~~~
No entry is made to the events history
~~~

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Fixes #8021

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?
After change:

**User VM**

- Destroy a HA-enabled VM on KVM hypervisor host using virsh
- Observe events for the VM

```
root@pr8601-t6928-kvm-ubuntu22-kvm1:~# date
Thu Jun 13 05:14:13 AM UTC 2024
root@pr8601-t6928-kvm-ubuntu22-kvm1:~# virsh destroy i-2-14-VM
Domain 'i-2-14-VM' destroyed
```
image

**VR**

- Destroy a HA-enabled VR on KVM hypervisor host using virsh
- Observe events for the VR

```
root@pr8601-t6928-kvm-ubuntu22-kvm2:~# date
Thu Jun 13 05:18:12 AM UTC 2024
root@pr8601-t6928-kvm-ubuntu22-kvm2:~# virsh destroy r-13-VM
Domain 'r-13-VM' destroyed
```
image

### CODE REVIEW

File: services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxy.java [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (DaanHoogland) [inline services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxy.java:1]:
```suggestion
```

**Review by DaanHoogland — APPROVED** [review]
clgtm, just wondering if we could call the methods just `startForHA(..)` instead of type specific names? I know the services to call are different, so it's only cosmetic.

**Comment by shwstppr** [comment]
@DaanHoogland separate methods are called allow publishing of events of different types. Initially, I did the same.
Other way to allow publishing of correct events could be manually calling `ActionEventUtils.onStarted/onCompleted/onError`

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
> @DaanHoogland separate methods are called allow publishing of events of different types. Initially, I did the same. Other way to allow publishing of correct events could be manually calling `ActionEventUtils.onStarted/onCompleted/onError`

not sure if we understand each other @shwstppr , couldn't `routerService.startRouterForHA(vm, params, planner);` be just called `routerService.startForHA(vm, params, planner);` and `consoleProxyManager.startProxyForHA(vm, params, planner);` be called `consoleProxyManager.startForHA(vm, params, planner);` and so on?

**Comment by DaanHoogland** [comment]
> oh. I thought you meant using a single method. That can be changed. I based the on the existing methods in the service, eg: https://github.com/apache/cloudstack/pull/9202/files#diff-94dd8d188fcf167c1fe8196f66e9817665641c95eda7f3c8fa03c2876ad1a1a2R60

I was just seeing the possiboility for some polimorphism. retrieve the interface and run a generic method on it. I am not sure if this will give us much cleanup or maintainability in this case though. Do as you see fit.

**Comment by shwstppr** [comment]
yes, currently it won't benefit. Will try to explore a generic interface like BaseVmManager in the coming weeks

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**